packages/protocol/contracts/rrp/authorizers/RequesterRrpAuthorizer.sol

```diff
@@ -23,10 +70,19 @@ abstract contract RequesterRrpAuthorizer is
     function isAuthorized(
         bytes32 requestId, // solhint-disable-line no-unused-vars
         address airnode,
-        bytes32 endpointId, // solhint-disable-line no-unused-vars
+        bytes32 endpointId,
         address sponsor, // solhint-disable-line no-unused-vars
         address requester
     ) external view override returns (bool) {
-        return userIsWhitelisted(bytes32(abi.encode(airnode)), requester);
+        return
+            userIsWhitelisted(deriveAdminnedId(airnode, endpointId), requester);
+    }
+
+    function deriveAdminnedId(address airnode, bytes32 endpointId)
+        internal
+        view
```

Comment:
Isn't this a pure function since it's not reading from state?

Reply:
It is